cmds/installcommand/installcommand.go

```diff
@@ -121,13 +122,10 @@ func main() {
 		debug(string(out))
 	}
 
-	cmd = exec.Command(destFile)
+	args := append([]string{form.cmdName}, form.cmdArgs...)
+	err = syscall.Exec(destFile, args, os.Environ())
```

Comment:
This is an attractive but unsafe idea. There are no system calls that modify a single process state that it is safe to call. The reason is that Go applications can consist of an ensemble of processes, and what you're doing here is making one of that ensemble vanish. You're going around behind the back of the runtime and replace one process with a another. 

I realize it may work but it's still not safe over the long time. They reserve the right to do pretty much anything to the runtime. 

This is also why you can't safely use system calls like unshare or, in fact, any system call that changes single process state. Line by line, the process running the code can change. You should see some of the nasty hacks they had to do for docker to get process state correct in the very early going.

So, cool idea, but to make it work is going to require some heavy lifting in the runtime. You can't do it this way.